### DISCUSSION TITLE
Troubleshooting Dynamic ContentCompatibility Update(player-sdk)

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#188] [Bugs]
Hi,

We recently added a feature in the `release-6.0.0` branch of the [sunbird-player-sdk](https://github.com/project-sunbird/sunbird-player-sdk/tree/release-6.0.0 ) that allows for changing the contentCompatibility value.
Reference:- [update contentComptabailityLevel](https://github.com/orgs/Sunbird-Knowlg/discussions/175)
However, this dynamic update seems to be causing problems in the Sunbird desktop app, leading to an Injection Token Error.
![MicrosoftTeams-image](https://github.com/Sunbird-Knowlg/Community/assets/82456953/f5c1307d-6289-4dba-a2f9-6aa0d130276a)

Despite working fine in the Sunbird Ed-portal and our local setup of the Sunbird desktop app, we're struggling to pinpoint the exact cause.
To investigate further, we propose creating a branch from `release-5.1.0` where we update the contentCompatability level in a non-dynamic manner. This will enable us to conduct testing by generating some alpha or beta versions of the sdk-player.
cc: 
@pallakartheekreddy @HarishGangula @rajeevsathish
@sajeshkayyath @PoojaJManjooran @vaibhavbhuva @abhishekpnt

**Comment by HarishGangula** [comment]
@rajnishdargan  Can if it is working locally not in when packaged means that when you do angular prod build for desktop app might be the issue please do prod build version in local and check

**Reply by rajnishdargan** [reply]
Hi @HarishGangula 
It also functions locally in the production build.

I've been investigating the issue further and I tried to use the non dynamic way of updating the compatabiltyLevel by adding a new variable and new method to check the questionsetCompatability in sunbird-player-sdk like below.

`playerQuestionsetCompatibiltyLevel = 6;
`
```
checkQuestionsetCompatibility(currentCompatibilityLevel: number) {
    if (currentCompatibilityLevel > this.playerQuestionsetCompatibiltyLevel) {
      const compatibilityError = new Error();
      compatibilityError.message = `Player supports ${this.playerQuestionsetCompatibiltyLevel}
      but content compatibility is ${currentCompatibilityLevel}`;
      compatibilityError.name = 'contentCompatibily';
      return { error: compatibilityError, isCompitable: false };
    } else {
      return { error: null, isCompitable: true };
    }
  }
```
This change resolves the Injection Token Error, but issues persist specifically in the End Page when building the desktop application. No errors are  in the console or network tab.

<img width="960" alt="image" src="https://github.com/Sunbird-Knowlg/Community/assets/82456953/c747cc79-925f-4cf5-a721-b9b039d17ac1">

Locally, the modified sunbird-player-sdk functions also works correctly in both development and production builds. However, issues arise only in the production build when generating the .exe file for the desktop application.

I suspect that there may be an issue occurring either due to an Angular version mismatch or because quml-player is utilizing version 6.0.0 of the SDK, while other players like video-player are using version 5.1.0.

Additionally, I mistakenly published version [6.0.1](https://www.npmjs.com/package/@project-sunbird/sunbird-player-sdk-v9/v/6.0.1) of the player SDK. Could you please delete this version? 

cc:
@pallakartheekreddy 
@rajeevsathish @abhishekpnt
@sajeshkayyath @PoojaJManjooran @vaibhavbhuva

**Reply by HarishGangula** [reply]
@rajnishdargan  can we upgrade the player sdk angular version and then check it once with the build

**Reply by pallakartheekreddy** [reply]
@rajnishdargan As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
